test/functional/api/vim_spec.lua

```diff
@@ -81,6 +81,26 @@ describe('api', function()
     end)
   end)
 
+  describe('nvim_execute_lua', function()
+    it('works', function()
+      meths.execute_lua('vim.api.nvim_set_var("test", 3)', {})
+      eq(3, meths.get_var('test'))
+
+      eq(17, meths.execute_lua('a, b = ...\nreturn a + b', {10,7}))
+    end)
+
+    it('reports errors', function()
+      meths.execute_lua('vim.api.nvim_set_var("test", 3)', {})
+      eq({false, 'Error loading lua: [string "<nvim>"]:1: '..
+                 "'=' expected near '+'"},
+         meth_pcall(meths.execute_lua, 'a+*b', {}))
+
+      eq({false, 'Error executing lua: [string "<nvim>"]:1: '..
+                 "attempt to call global 'bork' (a nil value)"},
+         meth_pcall(meths.execute_lua, 'bork()', {}))
+    end)
```

Comment:
what error? lua just adds or subtracts `nil` as it sees fit. Maybe, if there any valid msgpack Object that can't be translated to valid lua, but I can't think of any...